ParkingZoneApp.Tests/Controllers/Admin/ParkingZoneControllerUnitTests.cs

```diff
@@ -0,0 +1,311 @@
+            };
+
+            _controller.ModelState.AddModelError(key, errorMessage);
+    
+            //Act
+            var result = _controller.Create(createVM);
+
+            //Assert
+            var model = Assert.IsType<ViewResult>(result).Model;
+            Assert.IsAssignableFrom<CreateVM>(model);
+            Assert.False(_controller.ModelState.IsValid);
+            Assert.Equal(JsonSerializer.Serialize(createVM), JsonSerializer.Serialize(model));
+        }
+
+        [Fact]
+        public void GivenValidCreateVM_WhenGetCreateIsCalled_ReturnsViewResult()
+        {
+            //Arrange
+            _parkingZoneServiceMock
+                    .Setup(x => x.GetById(parkingZone.Id))
+                    .Returns(parkingZone);
+
+            //Act
+            var result = _controller.Create();
+
+            //Assert
+            Assert.IsAssignableFrom<ViewResult>(result);
+            Assert.IsType<ViewResult>(result);
+        }
+        #endregion
+
+        #region Edit
+        [Fact]
+        public void GivenValidIDAndEditVM_WhenPostEditIsCalled_ThenModelStateIsTrueReturnsRedirectToIndex()
+        {
+            //Arrange
+            EditVM editVM = new EditVM()
+            {
+                Id = parkingZone.Id,
+                Name = parkingZone.Name,
+                Address = parkingZone.Address,
+                CreatedDate = parkingZone.CreatedDate,
+            };
+
+            _parkingZoneServiceMock
+                    .Setup(x => x.GetById(parkingZone.Id))
+                    .Returns(parkingZone);
+
+            //Act
+            var result = _controller.Edit(parkingZone.Id, editVM);
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.IsType<RedirectToActionResult>(result);
+            Assert.True(_controller.ModelState.IsValid);
+            _parkingZoneServiceMock.Verify(x => x.GetById(parkingZone.Id), Times.Once);
+            _parkingZoneServiceMock.Verify(x => x.Update(parkingZone), Times.Once);
+        }
+
+        [Fact]
+        public void GivenValidId_WhenPostEditIsCalled_ThenModelStateIsTrueReturnDbUpdateConcurrencyException()
+        {
+            //Arrange
+            _parkingZoneServiceMock
+                    .Setup(x => x.Update(parkingZone))
+                    .Throws<DbUpdateConcurrencyException>();
+
+            //Act
+            var result = _controller.Edit(parkingZone.Id, new EditVM());
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.IsType<NotFoundResult>(result);
+            _parkingZoneServiceMock.Verify(x => x.GetById(parkingZone.Id), Times.Once);
+            _parkingZoneServiceMock.Verify(x => x.Update(parkingZone), Times.Never);
+        }
+
+        [Theory]
+        [InlineData("Name", "Required")]
+        [InlineData("Address", "Required")]
+        public void GivenValidId_WhenEditIsCalled_ThenModelStateIsFalseReturnsViewResult(string field, string errorMessage)
+        {
```

Comment:
in the controller we dealed to move something into the ModelState Scope, you should move service and mapping into model state scope

Reply:
I don't get it what are you trying to say?

Reply:
YES you did it! But you did it for only Create, What about the EDIT POST?

Reply:
See you screenshoted Create action, we are staying in #region EDIT

Reply:
![image](https://github.com/dilmurod-rozikov/ParkingZone/assets/141717735/fffb4895-9236-426e-b26b-69d261a4229c)
it was in the right place already